test/core/primitives/primitives_codec_test.cpp

```diff
@@ -42,31 +50,27 @@ class Primitives : public testing::Test {
 
  protected:
   /// block header and corresponding scale representation
-  BlockHeader block_header_{{1},   // buffer: parent_hash
+  BlockHeader block_header_{createHash({0}),   // parent_hash
                             2,     // number: number
-                            {3},   // buffer: state_root
-                            {4},   // buffer: extrinsic root
+                            createHash({1}),   // state_root
+                            createHash({2}),   // extrinsic root
                             {5}};  // buffer: digest;
   Buffer encoded_header_{
-      4, 1,                    // {1}
-      2, 0, 0, 0, 0, 0, 0, 0,  // 2 as uint64
-      4, 3,                    // {3}
-      4, 4,                    // {4}
-      4, 5                     // {5}
+      128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```

Comment:
please comment which parts of array is what. Before it contained comments with explanations